src/stream.rs

```diff
@@ -11,44 +11,57 @@ use std::{
 
 use tokio::io::{AsyncRead, AsyncWrite, ReadBuf};
 
-/// Stream, either plain TCP or TLS.
+/// A stream that might be protected with TLS.
 #[pin_project(project = StreamProj)]
-pub enum Stream<S, T> {
+pub enum MaybeTlsStream<S> {
```

Comment:
I think we should add `#[non_exhaustive]`.

Reply:
Definitely, I forgot this one. Thanks for noting 🙇